flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchApiCallBridge.java

```diff
@@ -93,6 +93,18 @@ default RequestIndexer createBulkProcessorIndexer(
 			numPendingRequestsRef);
 	}
 
+	/**
+	 * Creates a {@link RequestIndexer} that is able to work with {@link BulkProcessor} binary compatible.
+	 */
+	default ElasticsearchFailureHandlerIndexer createFailureHandlerIndexer(
+			boolean flushOnCheckpoint,
+			AtomicLong numPendingRequestsRef) {
```

Comment:
the fact that we require passing in the `numpendingRequestsRef` hints that we might need to refactor this a bit ..

Reply:
Perhaps the responsibility of keeping track of the number of in-flight requests should be tracked by a `BulkProcessorWrapper`.